backend/src/main/java/com/github/NicolaiKopka/users/AccountService.java

```diff
@@ -35,6 +36,23 @@ public MyUser registerUser(RegisterData registerData) {
                             .roles(List.of("user"))
                             .build();
 
-        return myUserRepo.save(newUser);
+        MyUser setupUser = myUserRepo.save(newUser);
+
+        UserFavoritesSaveObject saveObject = new UserFavoritesSaveObject();
+        saveObject.setUserId(setupUser.getId());
+        userFavoritesRepo.save(saveObject);
+
+        return setupUser;
+    }
+
+    //after deploy section to modify DB on next login with possible changes in backend handling
+    public void modifyDBOnLogin(MyUser user) {
+
+        //ensures every user has a related UserFavoriteSaveObject or has one created on next login
+        if(userFavoritesRepo.findByUserId(user.getId()).isEmpty()) {
```

Comment:
Ah i see ;)
i solved this problem a little differently. in the relevant service i have a get UserFavoritesObject method which creates an empty one if nothing is found. this is very easy with `repo.findById(id).orElse(new UserFav...)`